src/Microsoft.AspNetCore.Blazor.Build/targets/Blazor.MonoRuntime.targets

```diff
@@ -0,0 +1,305 @@
+      <BlazorIndexHtml>$(BlazorWebRootPath)$(BlazorIndexHtmlName)</BlazorIndexHtml>
+    </PropertyGroup>
+
+    <ItemGroup Label="BlazorItemSpecs">
+      <BlazorItemSpec Include="@(MonoAsmjsFile)">
+        <TargetOutputPath>$(ProjectDir)$(OutputPath)$(BaseBlazorRuntimeAsmjsOutputPath)%(FileName)%(Extension)</TargetOutputPath>
+        <Type>AsmJs</Type>
+        <IsStatic>true</IsStatic>
+      </BlazorItemSpec>
+      <BlazorItemSpec Include="@(MonoWasmFile)">
+        <TargetOutputPath>$(ProjectDir)$(OutputPath)$(BaseBlazorRuntimeWasmOutputPath)%(FileName)%(Extension)</TargetOutputPath>
+        <Type>WebAssembly</Type>
+        <IsStatic>true</IsStatic>
+      </BlazorItemSpec>
+      <BlazorItemSpec Include="@(BlazorJsFile)">
+        <TargetOutputPath>$(ProjectDir)$(OutputPath)$(BaseBlazorJsOutputPath)%(FileName)%(Extension)</TargetOutputPath>
+        <Type>BlazorRuntime</Type>
+        <IsStatic>true</IsStatic>
+      </BlazorItemSpec>
+    </ItemGroup>
+
+    <PropertyGroup Label="Intermediate output paths">
+
+      <!-- /obj/debug/netstandard2.0/blazor -->
+      <BlazorIntermediateOutputPath>$(ProjectDir)$(IntermediateOutputPath)$(BaseBlazorIntermediateOutputPath)</BlazorIntermediateOutputPath>
+
+      <!-- /obj/debug/netstandard2.0/blazor/linker/ -->
+      <BlazorIntermediateLinkerOutputPath>$(BlazorIntermediateOutputPath)linker/</BlazorIntermediateLinkerOutputPath>
+      
+      <!-- /obj/debug/netstandard2.0/blazor/linker.marker.txt -->
+      <BlazorIntermediateLinkerResultFilePath>$(BlazorIntermediateOutputPath)linker.marker.txt</BlazorIntermediateLinkerResultFilePath>
+
+      <!-- /obj/debug/netstandard2.0/blazor/Index.html -->
+      <BlazorIndexHtmlOutputPath>$(BlazorIntermediateOutputPath)$(BlazorIndexHtmlName)</BlazorIndexHtmlOutputPath>
+    
+      <!-- /obj/debug/netstandard2.0/blazor/Descriptor.xml -->
+      <GeneratedBlazorLinkerDescriptor>$(BlazorIntermediateOutputPath)Descriptor.xml</GeneratedBlazorLinkerDescriptor>
+
+      <!-- /obj/debug/netstandard2.0/blazor/Descriptor.xml -->
+      <BlazorResolvedAssembliesOutputPath>$(BlazorIntermediateOutputPath)ResolvedAssemblies.txt</BlazorResolvedAssembliesOutputPath>
+
+    </PropertyGroup>
+    
+    <ItemGroup Label="Different outputs used through the build">
+      <ProducedLinkerDescriptor Include="$(GeneratedBlazorLinkerDescriptor)" />
+      <LinkerMarkerFileOutput Include="$(BlazorIntermediateLinkerResultFilePath)" />
+      <BlazorIndexHtmlFileOutput Include="$(BlazorIndexHtmlOutputPath)" />
+      <BlazorResolvedAssembliesOutput Include="BlazorResolvedAssembliesOutputPath" />
+    
+  </ItemGroup>
+   
+    <MakeDir Directories="$(BlazorIntermediateOutputPath)" />    
+    
+  </Target>
+
+  <Target Name="ResolveBinaryOutput" DependsOnTargets="CollectLinkerOutputs;GetBaseClassLibrary" />
+
+  <Target Name="CollectLinkerDescriptors"
+          Inputs="@(IntermediateAssembly);@(BlazorLinkerDescriptor);@(ProducedLinkerDescriptor)"
```

Comment:
If you have a list of files of inputs, you may need to have them and store that in a file. This can matter when something is **removed**.

Reply:
I've implemented incremental compilation properly to fix this